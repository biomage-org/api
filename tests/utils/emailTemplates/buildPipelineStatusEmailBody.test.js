const { ACCOUNT_ID, SUCCEEDED, FAILED } = require('../../../src/api.v2/constants');
const config = require('../../../src/config');
const buildPipelineStatusEmailBody = require('../../../src/utils/emailTemplates/buildPipelineStatusEmailBody');

const mockExperimentId = 'mock-experiment-id';
const mockUser = {
  name: 'Test User',
  email: 'test@example.com',
};

describe('buildUserInvitedNotRegisteredEmailBody', () => {
  it('Should build the correct message for SUCCEEDED status', () => {
    const emailParams = buildPipelineStatusEmailBody(mockExperimentId, SUCCEEDED, mockUser);
    expect(emailParams).toMatchSnapshot();
  });

  it('Should build the correct message for FAILED status', () => {
    const emailParams = buildPipelineStatusEmailBody(mockExperimentId, FAILED, mockUser);
    expect(emailParams).toMatchSnapshot();
  });

  it('Should not show additional retry info for HMS emails', () => {
    config.awsAccountId = ACCOUNT_ID.HMS;
    const emailParams = buildPipelineStatusEmailBody(mockExperimentId, FAILED, mockUser);
    expect(emailParams).toMatchSnapshot();
  });
});
