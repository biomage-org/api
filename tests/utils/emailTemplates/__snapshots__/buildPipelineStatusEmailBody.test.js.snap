// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildUserInvitedNotRegisteredEmailBody Should build the correct message for FAILED status 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@example.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              The Biomage Team
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              The Biomage Team
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Cellenics experiment status",
    },
  },
  "Source": "notification@biomage.net",
}
`;

exports[`buildUserInvitedNotRegisteredEmailBody Should build the correct message for SUCCEEDED status 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@example.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      The data processing pipeline has completed successfully and your data is now ready to explore:<br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a>
      Happy analysing! <br/><br/><br/>
              The Biomage Team
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      The data processing pipeline has completed successfully and your data is now ready to explore:<br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a>
      Happy analysing! <br/><br/><br/>
              The Biomage Team
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Cellenics experiment status",
    },
  },
  "Source": "notification@biomage.net",
}
`;

exports[`buildUserInvitedNotRegisteredEmailBody Should not show additional retry info for HMS emails 1`] = `
Object {
  "Destination": Object {
    "ToAddresses": Array [
      "test@example.com",
    ],
  },
  "Message": Object {
    "Body": Object {
      "Html": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
      "Text": Object {
        "Charset": "UTF-8",
        "Data": "
        <html>
        <head>
        </head>
        <body>
            <h3>Hello Test,</h3>
            <p>Thanks for using Cellenics! <br/>
              
      Unfortunately, when trying to run the analysis the data processing failed for your experiment: <br/>
      <a href=\\"scp.biomage.net/experiments/mock-experiment-id/data-processing\\">scp.biomage.net/experiments/mock-experiment-id/data-processing</a> <br/>
      We are working hard to identify and fix the issue. We will let you know as soon as your data is ready to explore. <br/>
      <br/><br/>
              <small> <br/> <br/> You can disable the notifications for this experiment when you start processing it again. </small>
            <p/>
        </body>
      </html>",
      },
    },
    "Subject": Object {
      "Charset": "UTF-8",
      "Data": "Cellenics experiment status",
    },
  },
  "Source": "alex_pickering@hms.harvard.edu",
}
`;
